tests/test_server.py

```diff
@@ -0,0 +1,155 @@
+"""
+Inventory API Service Test Suite
+
+Test cases can be run with the following:
+  nosetests -v --with-spec --spec-color
+  coverage report -m
+"""
+
+from flask_api import status
+from mock import MagicMock, patch
+from models import ProductInformation, DataValidationError, db
+import json
+import logging
+import os
+import server
+import unittest
+
+######################################################################
+#  Fixed Global Variables
+######################################################################
+DATABASE_URI = os.getenv('DATABASE_URI', 'sqlite:///db/test.db')
+# Data table entry names
+PROD_ID = 'prod_id'
+PROD_NAME = 'prod_name'
+NEW_QTY = 'new_qty'
+USED_QTY = 'used_qty'
+OPEN_BOXED_QTY = 'open_boxed_qty'
+RESTOCK_LEVEL = 'restock_level'
+RESTOCK_AMT = 'restock_amt'
+# Default ProductInformation property value
+DEFAULT_NEW_QTY = 0
+DEFAULT_USED_QTY = 0
+DEFAULT_OPEN_BOXED_QTY = 0
+DEFAULT_RESTOCK_LEVEL = -1
+DEFALUT_RESTOCK_AMT = 0
+# API paths
+PATH_INVENTORY = '/inventory'
+# Content type
+JSON = 'application/json'
+# Location header
+LOCATION = 'Location'
+
+######################################################################
+#  Test Cases
+######################################################################
+class TestInventoryServer(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """ Run once before all tests """
+        server.app.debug = False
+        #server.initialize_logging(logging.INFO)
+        server.app.config['SQLALCHEMY_DATABASE_URI'] = DATABASE_URI
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+    
+    def setUp(self):
+        """ Runs before each test """
+        server.init_db()
+        db.drop_all()
+        db.create_all()
+        ProductInformation(prod_id=1, prod_name='a', new_qty=1, used_qty=1, open_boxed_qty=1,
+                restock_level=10, restock_amt=10).save()
+        ProductInformation(prod_id=2, prod_name='b', new_qty=2, used_qty=2, open_boxed_qty=2,
+                restock_level=20, restock_amt=20).save()
+        self.app = server.app.test_client()
+
+    def tearDown(self):
+        db.session.remove()
+        db.drop_all()
+
+    def test_create_prod_info_BadRequestException(self):
```

Comment:
I'm not quite familiar with python naming convention, but is this a good practice here to mix camel casing with python-type-underscore? E.g. maybe use test_create_prod_info_bad_request_exception() ? Better check with Pylint to see what it says about the styling.

Reply:
Addressed